### PR TITLE
pybind/mgr: ceph osd status crash with ZeroDivisionError

### PR DESCRIPTION
**Pull request by NitzanMordhai**
If stats-update is called a second time within a second, get_rate() fails with
a 'divide by 0' error.
Change the check before the computation, taking into account the fact that
two f.p numbers may differ, but still their diff - when cast into an int - might
be zero.

Fixed: https://tracker.ceph.com/issues/53538
Signed-off-by: Nitzan Mordechai <nmordech@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [x] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/status/module.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def get_latest(self, daemon_type: str, daemon_name: str, stat: str) -> int:
 
     def get_rate(self, daemon_type: str, daemon_name: str, stat: str) -> int:
         data = self.get_counter(daemon_type, daemon_name, stat)[stat]
-        if data and len(data) > 1 and data[-1][0] != data[-2][0]:
+        if data and len(data) > 1 and int(data[-1][0] - data[-2][0]) != 0:
```
[the remark this answers is not in the document]
Comment (NitzanMordhai) [reply src/pybind/mgr/status/module.py]: done. thanks

**Comment by ronen-fr** [comment]
@NitzanMordhai : I made some changes to the PR message. You might want to consider the revised text
for the commit message.

**Comment by NitzanMordhai** [comment]
> @NitzanMordhai : I made some changes to the PR message. You might want to consider the revised text
> for the commit message.

@ronen-fr I copied the text changes into the commit

**Comment by sebastian-philipp** [comment]
have you tried using `mgr_util.get_time_series_rates` ?

https://github.com/ceph/ceph/blob/82219b3beae122f7b93360ca586ba46f4aee5b9a/src/pybind/mgr/mgr_util.py#L665

**Comment by NitzanMordhai** [comment]
> have you tried using `mgr_util.get_time_series_rates` ?
> 
> https://github.com/ceph/ceph/blob/82219b3beae122f7b93360ca586ba46f4aee5b9a/src/pybind/mgr/mgr_util.py#L665

@sebastian-philipp  the original function worked fine, the only change was the check before return, the conversion to int caused value lower then 1 to divide by zero

**Review by jdurgin — APPROVED** [review]

**Review by ronen-fr — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]
nit: s/Fixed/Fixes/ in the commit message

**Pull request merged**